### PR TITLE
executor, statistics: use FM sketch to estimate NDV

### PR DESCRIPTION
**Pull request by lamxTyler**
@shenli  @coocood  @hanfei1991 @winoros  PTAL

### CODE REVIEW

**Comment by hanfei1991** [comment]
add test for estimating ndv

File: statistics/fmsketch.go [hunk 1]
```diff
@@ -50,27 +56,34 @@ func (s *FMSketch) insertHashValue(hashVal uint64) {
...
 
-func buildFMSketch(values []types.Datum, maxSize int) (*FMSketch, int64, error) {
-	s := newFMSketch(maxSize)
+// InsertValue inserts a value into the FM sketch.
+func (s *FMSketch) InsertValue(value types.Datum) error {
+	bytes, err := codec.EncodeValue(nil, value)
+	if err != nil {
+		return errors.Trace(err)
+	}
 	h := fnv.New64a()
```
Comment (hanfei1991) [inline statistics/fmsketch.go]: You can new once and reset every time.

Comment (lamxTyler) [inline statistics/fmsketch.go]: You mean make it global or pass it an argument?

Comment (hanfei1991) [inline statistics/fmsketch.go]: make it as a field in FMSketch.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**